low-level/framework/base/global_config.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3.6
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+from cortx.utils.conf_store import Conf
+from framework.utils.conf_utils import (GLOBAL_CONF, SSPL_CONF,
+            CLUSTER, SERVER_NODES, MACHINE_ID, STORAGE,
+            ENCLOSURE_ID)
+from framework.base.sspl_constants import (DATACENTER01,
+                                           RACK01, SERVERNODE01,
+                                           CLUSTER01)
+
+Conf.load(SSPL_CONF, "yaml:///etc/sspl.conf")
+global_config = Conf.get(SSPL_CONF, "SYSTEM_INFORMATION>global_config_copy_url")
+Conf.load(GLOBAL_CONF, global_config)
+
+SRVNODE = Conf.get(GLOBAL_CONF, f'{CLUSTER}>{SERVER_NODES}')[MACHINE_ID]
```

Comment:
One discussion today with Ujjwal altered the whole meaning of having this PR. We want to **avoid duplicated code**, so we pulling common code in one file. However, config values here pulled once using confstor apis and then made available throughout life of the sspl service.

**Problem is what if value changes dynamically ?** Confstor apis would reflect changed values every time we call it, while here with this approach it wont. 

Now how do we address both ? Again a wrapper functions in this file to call confstor api kv pair we need, instead better is existing code then, which directly calls confstor api per sensor.

I dont know how to tackle this yet, better we think through & then implement. Originally we were limiting to new sensor code only, then were planning to extend later for existing sensors (since we would know the implications for new approach too)

Reply:
how about one function in here to fetch common type of configs using confstore apis, like

cluster_id, site_id, rack_id, machine_id= get_resource_location(fetch_cluster_id=false, fetch_site_id=false, fetch_rack_id=false, fetch_machine_id=false)

With filters to fetch any required id even, like
clusterid, siteid, , = get_resource_location(true,true, false, false)

Same way other functions we can have more functions in future,  to fetch more global configs in group together, like for storage_set, servers_in_storageset, enclosures_in_storage_set

These functions call per sensor would be needed to fetch required configs then !

**This is just one proposal, feel to add better ones**

Reply:
To generalize the code, I am thinking we can keep two dict.
one global dict will map the key with val as a tuple of req fields.
For eg:-
        config_dict = {
            "cluster_id": (CLUSTER, SRVNODE, 'site_id')
        }
And one more dict to have default values of the key
for eg:- 
        default_val = {
            "cluster_id": 'CC01'
        }		


And a function which will take a config_list required by diff sensors
and will process the request and will return the response dict.
	def fetch_config_from_global_dict(self, config_list):
        response = {}
        default = ""
        for key in config_list:
            if key not in config_dict.keys():
                return
            val = config_dict[key]
            if key in default_val.keys():
                default = default_val[key]
            req_str =  '>'.join(val)
            global_val = Conf.get(GLOBAL_CONF, req_str, default)
            response[key] = global_val
        return response

But with this approach we have to maintain 2 dicts and have to map the req fields by sensors to this dict.
@cdeshmukh @sandeepsgt  please review this approach, if it seems too complex, we can proceed with chetan's approach.

Reply:
Am not totally clear on this approach, we can discuss.

But definite take away is to keep the list of keys which can be requested, as flexible.

And we also want to take away the complexity & repeated code of requesting common/sspl config from consumer layers like sensors & actuators.

So we can two main functions
<return dict wih kv pairs> get_sspl_config(<set of keys as i/p>)
<return dict wih kv pairs> get_global_config(<set of keys as i/p>)

Now both these functions would take just keys (with hierarchies) and internally use confstor apis to fetch from respective backends (config index). So sensors/actuators will still have to import keys as its done today.

Reply:
we discussed this later & latest code reflects what was needed